File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.java
import java.util.ArrayList;
class Solution {
    public long[] solution(long[] numbers) {
        long[] answer = new long[numbers.length];
        ArrayList<Long> arr = new ArrayList<>();
        for (int i = 0; i < numbers.length; i++) {
            long num = numbers[i];
            
            if (num % 2 == 0) {
                arr.add(num + 1);
                continue;
            } else {
                long bit = (~num) & (num + 1);
                long target = num + (bit>>1);
                arr.add(target);
            }
        }
        
        for (int i = 0; i < numbers.length; i++) {
            answer[i] = arr.get(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.py
def solution(numbers):
    answer = []
    N = len(numbers)
    for i in range(N):
        num = numbers[i]
        if (num % 2 == 0):
            answer.append(num+1)
            continue
        else:
            tmp = (~num) & (num+1)
            target = num + (tmp>>1)
            answer.append(target)
    return answer